src/main/java/com/minecolonies/coremod/client/gui/townhall/WindowMainPage.java

```diff
@@ -64,111 +101,376 @@ public WindowMainPage(final BuildingTownHall.View building)
+        {
+            Network.getNetwork().sendToServer(new ColonyTextureStyleMessage(building.getColony(), TEXTURE_PACKS.get(dropDownList.getSelectedIndex())));
+        }
+    }
+
+    /**
+     * Toggle the dropdownlist with the selected index to change the texture of the colonists.
+     *
+     * @param dropDownList the toggle dropdown list.
+     */
+    private void toggleNameFile(final DropDownList dropDownList)
+    {
+        if (dropDownList.getSelectedIndex() != initialNamePackIndex)
+        {
+            Network.getNetwork().sendToServer(new ColonyNameStyleMessage(building.getColony(), building.getColony().getNameFileIds().get(dropDownList.getSelectedIndex())));
+        }
+    }
+
+    /**
+     * Called when the dropdownList changed.
+     *
+     * @param dropDownList the list.
+     */
+    private void onDropDownListChanged(final DropDownList dropDownList)
+    {
+        Network.getNetwork().sendToServer(new TeamColonyColorChangeMessage(dropDownList.getSelectedIndex(), building));
+    }
+
+    /**
+     * Opens the banner picker window. BOWindow does not use blockui, so is started manually.
+     *
+     * @param button the trigger button
+     */
+    private void openBannerPicker(@NotNull final Button button)
+    {
+        Screen window = new WindowBannerPicker(building.getColony(), this, isFeatureUnlocked);
+        Minecraft.getInstance().execute(() -> Minecraft.getInstance().setScreen(window));
+    }
+
+    /**
+     * Reset the texture style.
+     */
+    private void resetTextureStyle()
+    {
+        Network.getNetwork().sendToServer(new ColonyTextureStyleMessage(building.getColony(), TEXTURE_PACKS.get(0)));
+    }
+
+    @Override
+    public void onUpdate()
+    {
+        super.onUpdate();
+        final Pane textPane = findPaneByID(DROPDOWN_TEXT_ID);
+        final Pane namePane = findPaneByID(DROPDOWN_NAME_ID);
+        final Pane resetButton = findPaneByID(BUTTON_RESET_TEXTURE);
+
+        if (isFeatureUnlocked.get())
+        {
+            textPane.enable();
+            namePane.enable();
+            textPane.show();
+            resetButton.hide();
+        }
+        else
+        {
+            textPane.disable();
+            namePane.disable();
+
+            if (!building.getColony().getTextureStyleId().equals("default"))
+            {
+                resetButton.show();
+                textPane.hide();
+            }
+            else
+            {
+                textPane.show();
+            }
+
+            AbstractTextBuilder.TooltipBuilder hoverText = PaneBuilders.tooltipBuilder().hoverPane(textPane);
+            hoverText.append(Component.translatable("com.minecolonies.core.townhall.patreon.textures")).paragraphBreak();
+            hoverText.appendNL(Component.empty());
+            hoverText.appendNL(Component.translatable("com.minecolonies.core.townhall.patreon")).paragraphBreak();
+            hoverText.build();
```

Comment:
It's builder so you can chain method calls :D
No need to paragraph break before build (few lines below too)